### ISSUE TITLE
Exception in thread "main" java.lang.NullPointerException: Cannot invoke "String.length()" because "s" is null

### ISSUE THREAD

**Issue by SpaceCondor** [SAP/cloud-sdk#858]
## Issue Description

When generating a client, the following error occurs:

```
Exception in thread "main" java.lang.NullPointerException: Cannot invoke "String.length()" because "s" is null
        at com.sun.codemodel.JExpr.quotify(JExpr.java:239)
        at com.sun.codemodel.JStringLiteral.generate(JStringLiteral.java:61)
        at com.sun.codemodel.JFormatter.g(JFormatter.java:350)
        at com.sun.codemodel.JReturn.state(JReturn.java:66)
        at com.sun.codemodel.JFormatter.s(JFormatter.java:386)
        at com.sun.codemodel.JBlock.generateBody(JBlock.java:448)
        at com.sun.codemodel.JBlock.generate(JBlock.java:436)
        at com.sun.codemodel.JFormatter.g(JFormatter.java:350)
        at com.sun.codemodel.JBlock.state(JBlock.java:464)
        at com.sun.codemodel.JFormatter.s(JFormatter.java:386)
        at com.sun.codemodel.JMethod.declare(JMethod.java:460)
        at com.sun.codemodel.JFormatter.d(JFormatter.java:376)
        at com.sun.codemodel.JDefinedClass.declareBody(JDefinedClass.java:815)
        at com.sun.codemodel.JDefinedClass.declare(JDefinedClass.java:788)
        at com.sun.codemodel.JFormatter.d(JFormatter.java:376)
        at com.sun.codemodel.JFormatter.write(JFormatter.java:406)
        at com.sun.codemodel.JPackage.build(JPackage.java:438)
        at com.sun.codemodel.JCodeModel.build(JCodeModel.java:311)
        at com.sun.codemodel.JCodeModel.build(JCodeModel.java:301)
        at com.sap.cloud.sdk.datamodel.odatav4.generator.CodeModelClassGenerator.writeCodeModel(CodeModelClassGenerator.java:131)
        at com.sap.cloud.sdk.datamodel.odatav4.generator.CodeModelClassGenerator.writeClasses(CodeModelClassGenerator.java:125)
        at com.sap.cloud.sdk.datamodel.odatav4.generator.ODataToVdmGenerator.generate(ODataToVdmGenerator.java:113)
        at com.sap.cloud.sdk.datamodel.odatav4.generator.DataModelGenerator.execute(DataModelGenerator.java:679)
        at com.sap.cloud.sdk.datamodel.odatav4.generator.DataModelGeneratorCli.run(DataModelGeneratorCli.java:217)
        at com.sap.cloud.sdk.datamodel.odatav4.generator.DataModelGeneratorCli.main(DataModelGeneratorCli.java:199)
```

This is using the latest CLI version (3.69.0) and the following OData definition:

[ChannelAdvisor.txt](https://github.com/SAP/cloud-sdk/files/8903122/ChannelAdvisor.txt)

Trying to generate with the following command:

`java -jar odata-v4-generator-cli.jar -b "/v1/"`

**Comment by KavithaSiva** [comment]
Hi @SpaceCondor,

Thank you for reaching out to us. I am currently looking into this issue and will update here soon.

Regards,
Kavitha

**Comment by SpaceCondor** [comment]
Hey @KavithaSiva 

Thank you! If there's any additional information I can provide, let me know!

Conner

**Comment by KavithaSiva** [comment]
Hi @SpaceCondor,

While running the generator for your specification file I also get errors logged saying that there are issues in the document.
I also found that it is the `ShippingOverride` entity that is causing the problem.

```xml

**Comment by KavithaSiva** [comment]
Hi @SpaceCondor,

While running the generator for your specification file I also get errors logged saying that there are issues in the document.
I also found that it is the `ShippingOverride` entity that is causing the problem.

If you remove this `Entity` and it's references from your specification file, the generation goes through successfully.
Remove these lines:
```xml
<EntityType Name="ShippingOverride">
    <Property Name="Sku" Type="Edm.String"/>
    <Property Name="Quantity" Type="Edm.Int32" Nullable="false" />
    <Property Name="DistributionCenterID" Type="Edm.Int32" Nullable="false" />
    <Property Name="ShippingCarrier" Type="Edm.String" />
    <Property Name="ShippingClass" Type="Edm.String" />
    <Property Name="TrackingNumber" Type="Edm.String" />
    <Property Name="SellerFulfillmentID" Type="Edm.String" />
    <Property Name="ShippedDateUtc" Type="Edm.DateTimeOffset" />
    <Property Name="ApplyShippingRules" Type="Edm.Boolean" Nullable="false" />
</EntityType>

<Action Name="Create" IsBound="true">
    <Parameter Name="bindingParameter" Type="Collection(ChannelAdvisor.RestApi.V1.Models.Order)" />
    <Parameter Name="Order" Type="ChannelAdvisor.RestApi.V1.Models.Order" />
    <Parameter Name="ShippingOverrides" Type="Collection(ChannelAdvisor.RestApi.V1.Models.ShippingOverride)" />
    <ReturnType Type="ChannelAdvisor.RestApi.V1.Models.Order" />
</Action>
```

I will get back to you after further investigation on what exactly is wrong in this Entity. ( Adding of the key to the entity didn't solve the issue)

Regards,
Kavitha

**Comment by SpaceCondor** [comment]
Hey @KavithaSiva,

Thanks for the input! I was able to generate the code.

However, I have another note -- the 'Package' type actually causes issues since that is a reserved word in Java.

Thanks,
Conner

**Comment by KavithaSiva** [comment]
Hi @SpaceCondor,

Apologies for the delayed response. Thanks for bringing the issue with the reserved keyword in the generator, we have drafted a backlog item to fix this issue. I will update you as soon as this issue is fixed.
But for now, as a workaround, you can update the places where the classname (`Package`) is being used as a variable name in the generated code.

For the other original issue of the `NullPointerException` we are still working on a fix, I will update you soon with the details.

Regards,
Kavitha

**Comment by MatKuhr** [comment]
I just stumbled over this, I think we forgot to update here. As far as I can tell the NPE and the issue regarding the reserved keywords have been fixed for quite some time, in version 4.2 or 4.3 or something.

I don't think entity parameters in actions are supported yet though (other than the binding parameter in case of bound actions). Still, closing this as the original issue is solved I think.
